presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -301,19 +304,29 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                 else {
                     connectorPartitionHandles = ImmutableList.of(NOT_PARTITIONED);
                 }
+
+                Map<Integer, Node> partitionToNode = nodePartitionMap.getPartitionToNode();
                 stageSchedulers.put(stageId, new FixedSourcePartitionedScheduler(
                         stage,
                         splitSources,
                         plan.getFragment().getStageExecutionStrategy(),
                         schedulingOrder,
-                        nodePartitionMap,
+                        IntStream.range(0, partitionToNode.size())
+                                .mapToObj(taskId -> {
+                                    Node node = partitionToNode.get(taskId);
+                                    verify(node != null);
+                                    return node;
+                                })
+                                .collect(toImmutableList()),
```

Comment:
Looking at the old `FixedSplitPlacementPolicy.allNodes` implementation, it appears like order doesn't matter.

If this is indeed the case, you can do `ImmutableList.copyOf(partitionToNode.values())`

Reply:
See comment https://github.com/prestodb/presto/pull/11693/files/58bd816641944672eec66b186f39902ad8d376e5..2ebdd45f985f00491ecb4638fb179f4f2ea3cf34#r233438798

Reply:
@haozhun : `FixedSplitPlacementPolicy.allNodes` doesn't care about order. However, the `taskId` cares about order. Since this is used as `partitionId` the remote source.